assets/js/modules/adsense/components/dashboard/AdBlockingRecoverySetupCTAWidget.test.js

```diff
@@ -96,7 +96,7 @@ describe( 'AdBlockingRecoverySetupCTAWidget', () => {
 			},
 		] );
 		registry.dispatch( CORE_USER ).setReferenceDate( referenceDate );
-		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );
+		registry.dispatch( CORE_USER ).receiveGetDismissedPrompts( {} );
```

Comment:
For point 1. I added a test to the `test.each` batch that checks the notification is hidden by an `expiry` in the future. The selector and the widget do not actually enforce dismissal based on the count, it will only be dismissed based on an expiry of 0 or in the future. Therefore in theory you could have a dismissal count above 3 and have the widget render although you wouldn't be able to get to this count state due to the `handleDismissClick` function.

For point 2. I added a new test in the `CTA actions` block which sets the count to 2 for the mock widget and checks for the presence of the `Don’t show again` button in the CTA.